src/jit/importer.cpp

```diff
@@ -4039,8 +4039,33 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
-            case NI_MathF_Round:
-            case NI_Math_Round:
+#ifdef FEATURE_HW_INTRINSICS
+            case NI_System_Math_FusedMultiplyAdd:
+            case NI_System_MathF_FusedMultiplyAdd:
+            {
+#ifdef _TARGET_XARCH_
+                if (compSupports(InstructionSet_FMA))
+                {
+                    // We are constructing a chain of intrinsics similar to:
+                    //    return FMA.MultiplyAddScalar(
+                    //        Vector128.CreateScalar(x),
+                    //        Vector128.CreateScalar(y),
+                    //        Vector128.CreateScalar(z)
+                    //    ).ToScalar();
+
+                    GenTree* op3 = gtNewSimdHWIntrinsicNode(TYP_SIMD16, impPopStack().val, NI_Base_Vector128_CreateScalarUnsafe, callType, 16);
```

Comment:
Another option might be to handle this like we do the other `System.Math` methods as a `CORINFO_INTRINSIC_FusedMultiplyAdd`.

This would also allow us to add value numbering support, and we could then swap this to the chain of HWIntrinsics in lowering... However, that also requires more changes and wouldn't work as well if we end up providing a non CRT implementation of `FusedMultiplyAdd` in the future.

I think a better option might be to add value numbering support for named intrinsics in general (which I think would require some refactoring of `GenTreeJitIntrinsic` to actually carry the intrinsic id -- currently it is only carried by `GenTreeHWIntrinsic`, since `GenTreeSimdIntrinsic` doesn't used named intrinsics and just carries the SimdIntrinsicId instead).... Thoughts?

Reply:
(the above would mean we carry `NI_System_Math_FusedMultiplyAdd` through value numbering and change it out in lowering for the appropriate HWIntrinsic chain; it might also make it easier to add value numbering support for HWIntrinsic and SimdIntrinsic nodes in the future, if we decided to do so).